worldedit-bukkit/src/main/java/com/fastasyncworldedit/bukkit/regions/plotsquared/PlotSquaredFeature.java

```diff
@@ -176,15 +176,28 @@ public FaweMask getMask(Player player, MaskType type, boolean isWhitelist) {
             maskedRegion = new RegionIntersection(world, weRegions);
         }
 
-        return new FaweMask(maskedRegion) {
-            @Override
-            public boolean isValid(Player player, MaskType type) {
-                if (Settings.Done.RESTRICT_BUILDING && DoneFlag.isDone(finalPlot)) {
-                    return false;
-                }
-                return isAllowed(player, finalPlot, type);
+        return new PlotSquaredMask(maskedRegion, finalPlot);
+    }
+
+    private final class  PlotSquaredMask extends FaweMask {
```

Comment:
Duplicate space between class and PlotSquaredMask 😡

Reply:
Yeah the whole thing kinda messy haha

Reply:
heh
![huh](https://user-images.githubusercontent.com/11150076/226720101-48d5d5e2-9fc7-4cfd-8646-2a56dd5dfffd.png)

but on a serious note, while I don't expect the hashCode to collide, it might be bad to rely on it - maybe it is fine to rely on reference equality of the sets? While that's not specified anywhere, we could add a note to the method in PlotSquared...

Reply:
I suppose wrapping the set in a weak reference removes any possibility of memory "leak" as if the connected plots is reset we don't need the set anyway, we can just check the reference